src/node_crypto.h

```diff
@@ -73,6 +73,32 @@ struct MarkPopErrorOnReturn {
   ~MarkPopErrorOnReturn() { ERR_pop_to_mark(); }
 };
 
+template<typename T, void (*function)(T*)>
+struct FunctionDeleter {
+  void operator()(T* pointer) const { function(pointer); }
+  typedef std::unique_ptr<T, FunctionDeleter> ptr;
```

Comment:
Can you name this `Ptr`, `Pointer`, `UniquePointer` or whatever to make it clearer it's a type?